### ISSUE TITLE
Pasted image not prompting a dialog

### ISSUE THREAD

**Issue by JStyle21** [reorx/obsidian-paste-image-rename#25]
**Describe the bug**
Pasted image doesn't prompt a dialog, also turning on all attachments doesn't fix this.

**Environments (please complete the following information):**
- OS: Archlinux
- Use `[[Wikilinks]]`: True (also tried false)
- Debug info:
SYSTEM INFO:
	Obsidian version: v1.0.3
	Installer version: v1.0.3
	Operating system: #1 SMP PREEMPT_DYNAMIC Thu, 03 Nov 2022 18:01:58 +0000 6.0.7-arch1-1
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: dark
	Community theme: none
	Snippets enabled: 1
	Restricted mode: off
	Plugins installed: 2
	Plugins enabled: 2
		1: Markdown Formatting Assistant v0.4.0
		2: Advanced Tables v0.17.3

**To Reproduce**
1. Copy image to clipboard (i use Spectacle for this)
2. Paste image into an Obsidian note

**Expected behavior**
A dialog prompting to rename should appear upon pasting the image

**Comment by 0xswitch** [comment]
seems related to https://github.com/reorx/obsidian-paste-image-rename/blob/caf25cf58e04ea2f7206bd389dccedb8a6e12670/src/main.ts#L63

**Comment by merlinuwe** [comment]
I have occasionally (not always) the same issue. The link is set, but without showing the dialog before.

SYSTEM INFO:
	Obsidian version: v1.1.9
	Installer version: v1.1.9
	Operating system: Windows 10 Home 10.0.19045
	Login status: not logged in
	Insider build toggle: off
	Live preview: off
	Legacy editor: off
	Base theme: dark
	Community theme: Minimal v6.1.17
	Snippets enabled: 22
	Restricted mode: off
	Plugins installed: 60
	Plugins enabled: 47
		1: Vault Statistics v0.1.3
		2: Templater v1.16.0
		3: Tag Wrangler v0.5.5
		4: Tasks v1.20.0
		5: Symbols Prettifier v1.1.1
		6: Style Settings v0.4.12
		7: Sortable v0.2.6
		8: Scroll to Top Plugin v1.5.0
		9: Recent Files v1.3.5
		10: Paste image rename v1.5.0
		11: Pandoc Plugin v0.4.1
		12: Obsidian Enhancing Export v1.1.7
		13: Obsidian Camera v1.5.0
		14: Number Headings v1.10.0
		15: Minimal Theme Settings v6.1.1
		16: Media Extended v2.11.1
		17: Linter v1.10.1
		18: Janitor v1.0.7
		19: Homepage v2.6
		20: Find orphaned files and broken links v1.8.0
		21: File Tree Alternative Plugin v2.2.7
		22: Execute Code v1.5.0
		23: Excel to Markdown Table v0.4.0
		24: Editor Syntax Highlight v0.1.3
		25: Editing Toolbar v2.2.3
		26: Dataview v0.5.47
		27: Copy document as HTML v0.3.3
		28: Contextual Typography v2.2.4
		29: Completr v3.1.0
		30: Commander v0.4.9
		31: Calendar v1.5.10
		32: Buttons v0.4.19
		33: Auto Class v2.2.0
		34: Advanced URI v1.31.2
		35: Bulk Rename v0.4.4
		36: DB Folder v3.1.0
		37: Waypoint v1.4.0
		38: Folder Note v0.7.3
		39: Hover Editor v0.11.8
		40: Show Current File Path v0.5.2
		41: Reminder v1.1.15
		42: Text expand v0.11.2
		43: Icon Folder v1.5.1
		44: Excalidraw v1.8.7
		45: MAKE.md v0.5.6
		46: Custom File Explorer sorting v1.4.0
		47: File Explorer Note Count v1.2.0

**Comment by reorx** [comment]
@0xswitch this should not be the culprit unless the image is so large and the computer is so overloaded that it takes more than 1 second for the image to be created.

I want to ask all people in this issue: how large is the image when you encounter this issue? Do you feel significant latency for the image to be pasted?

**Comment by JStyle21** [comment]
@reorx 
I can't say for sure since i'm doing my own screenshots so it can be an area or the whole screen but mostly they are areas with the size of around 200kb, i also didn't notice any lag or anything like that.

**Comment by reorx** [comment]
@JStyle21 got it, are you still encountering this issue after upgrading to the latest version (1.6.0)?

**Comment by JStyle21** [comment]
@reorx 
Yes, issue is still there, just installed a fresh 1.6 and tried to paste an image, tried all attachments also didn't work.
If there is something else to try or if i'm not using it properly let me know but i couldn't get the dialog even if the image is pasted from the FS or dragged or any other combo i could think of.

**Comment by 0xswitch** [comment]
maybe

> @0xswitch this should not be the culprit unless the image is so large and the computer is so overloaded that it takes more than 1 second for the image to be created.
> 
> I want to ask all people in this issue: how large is the image when you encounter this issue? Do you feel significant latency for the image to be pasted?

maybe the image isnt't that large but it is stored on a remote server (obsidian vault stored on a smb server for example) and it takes much time to save it.

**Comment by JStyle21** [comment]
@0xswitch 
I'm only using my obsidian locally and it doesn't work...
Can you try with an image capture app like spectacle, greenshot etc.?

**Comment by reorx** [comment]
This spectacle <https://github.com/KDE/spectacle>? Seems that I need a
Linux environment to test this plugin.

Xiao Meng

**Comment by JStyle21** [comment]
Yes, that's spectacle but you can use greenshot or any other app that performs screenshot to clipboard

**Comment by diomat** [comment]
Hi,

I'm having the same bug in Windows 10. I installed the plugin for the first time. So it never worked beforehand (except in the sandbox vault. see below)

- I tried it with active and inactive option *Handle all attachments*
- When I try it in the Obsidian Sandbox vault the dialogue the dialog **works**!
- The sandbox vault is saved locally (C:\) my actual vault is saved on a mapped network drive. So it seems there is a problem with network drives
- Image sizes 500-800 kb

**Describe the bug**
A clear and concise description of what the bug is.

**Environments (please complete the following information):**
- OS: Windows 10
- Use `[[Wikilinks]]`: false
- Debug info:
SYSTEM INFO:
	Obsidian version: v1.1.9
	Installer version: v0.15.9
	Operating system: Windows 10 Pro 10.0.19045
	Login status: not logged in
	Insider build toggle: off
	Live preview: off
	Legacy editor: off
	Base theme: dark
	Community theme: Dracula + LYT
	Snippets enabled: 1
	Restricted mode: off
	Plugins installed: 63
	Plugins enabled: 38
		1: Linter v1.10.2
		2: Markdown prettifier v0.1.3
		3: Outliner v4.2.1
		4: Pandoc Plugin v0.4.1
		5: PDF Highlights v0.0.4
		6: Recent Files v1.3.5
		7: Remember cursor position v1.0.7
		8: Table of Contents v0.1.3
		9: Timeline v1.1.0
		10: Text Snippets v0.1.2
		11: Icons v0.3.0
		12: Hotkey Helper v0.3.15
		13: Admonition v9.2.0
		14: Annotator v0.2.10
		15: Autocomplete v0.8.0
		16: Calendar v1.5.10
		17: Checklist v2.2.12
		18: Reminder v1.1.15
		19: Dataview v0.5.55
		20: Kroki v1.2.3
		21: Advanced Tables v0.18.1
		22: Style Settings v1.0.0
		23: Dynamic Highlights v0.3.1
		24: Commits v0.2.2
		25: Activity History v0.1.4
		26: Daily Activity v0.4.0
		27: Rollover Daily Todos v1.1.3
		28: Omnisearch v1.11.1
		29: Mousewheel Image zoom v1.0.13
		30: Obsidian Graphviz v1.0.5
		31: Excel to Markdown Table v0.4.0
		32: Obsidian TODO | Text-based GTD v0.2.7
		33: TimeStamper v1.3.0
		34: Various Complements v7.3.0
		35: Meld Encrypt v2.0.4
		36: Advanced Cursors v0.5.1
		37: Text Extractor v0.4.1
		38: Paste image rename v1.6.1

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.


**To Reproduce**
Steps to reproduce the behavior:
1. Take Screenshot with ShareX or use the Firefox feature to create image of website (right click on website)
2. Paste image into Obsidian
3. No dialog to rename image appears

**Expected behavior**
Dialogue to rename image

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Comment by Shiv-Patil** [comment]
I wanted to add that on my system (linux - zorin), the `file.stat.ctime` property is `0`. Why? No idea. Probably because the filesystem doesn't store that information or the api does not expose it.
This results in the plugin not prompting the dialog at all, as the `timeGapMs` will ALWAYS be more than 1 second - as it is essentially the current time (due to `ctime` being 0). A workaround for this might be using the `mtime` in the case when `ctime` is `0`.
